woocommerce/woocommerce
# normalize IDs for form fields in Checkout

With additional fields, we use the whole field key as a form ID, which will always have a `/` separator, the issue here is that html ID don't support `/` and using `querySelector` will throw an error. For that, we normalize the `/` to `-`.

### Changes proposed in this Pull Request:





Closes https://github.com/woocommerce/woocommerce/issues/47635



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Add the following snippet somewhere (or using a snippet plugin):
```php
add_action(
	'woocommerce_init',
	function () {
		woocommerce_register_additional_checkout_field(
			array(
				'id'       => 'plugin-name/test-field',
				'label'    => 'Test field',
				'location' => 'additional',
				'type'     => 'checkbox',
			)
		);
	}
);
```
2. Add an item to cart and go to Checkout.
3. Using console, run the following code: `document.querySelector('#additional-information-plugin-name-test-field')`.
4. There should be no errors.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 
Replace forward slashes in additional fields IDs with hyphens. 

#### Comment 

</details>
